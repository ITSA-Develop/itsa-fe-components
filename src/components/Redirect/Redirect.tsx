import { useEffect } from 'react';

export interface IRedirectProps {
	redirect: string;
}

export const Redirect = ({ redirect }: IRedirectProps) => {
	useEffect(() => {
		if (redirect && redirect.trim()) {
			window.location.assign(redirect);
		}
	}, [redirect]);
	return null;
};
