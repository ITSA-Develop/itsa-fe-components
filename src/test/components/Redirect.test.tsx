import { render } from '@testing-library/react';
import { Redirect } from '../../components/Redirect/Redirect';

beforeAll(() => {
	globalThis.location = {
		...window.location,
		assign: vi.fn(),
	};
});

describe('Redirect Component', () => {
	const redirectUrl = 'https://example.com';
	it('renders footer snapshot', () => {
		const { container } = render(<Redirect redirect={redirectUrl} />);
		expect(container).toMatchSnapshot();
	});

	it('should not redirect if no URL is provided', () => {
		render(<Redirect redirect="" />);
		expect(globalThis.location.assign).not.toHaveBeenCalled();
	});
});

afterEach(() => {
	vi.clearAllMocks();
});
