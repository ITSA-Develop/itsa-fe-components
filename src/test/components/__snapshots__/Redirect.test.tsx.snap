// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Redirect Component > renders footer snapshot 1`] = `<div />`;
